app/react/src/server/framework-preset-react.ts

```diff
@@ -28,17 +28,17 @@ const context = storybookReactDirName.includes('node_modules')
   ? path.join(storybookReactDirName, '../../') // Real life case, already in node_modules
   : path.join(storybookReactDirName, '../../node_modules'); // SB Monorepo
 
-export async function babelDefault(config: TransformOptions) {
-  let reactVersion;
+const hasJsxRuntime = () => {
   try {
-    // eslint-disable-next-line global-require,import/no-dynamic-require
-    const reactPkg = require(require.resolve('react/package.json', { paths: [context] }));
-    reactVersion = reactPkg.version;
-  } catch {
-    logger.warn('Unable to determine react version');
+    require.resolve('react/jsx-runtime', { paths: [context] });
+    return true;
+  } catch (e) {
+    return false;
   }
-  const presetReactOptions =
-    reactVersion && semver.gte(reactVersion, '16.14.0') ? { runtime: 'automatic' } : {};
```

Comment:
Now that you don't need `semver`, you can remove it from https://github.com/storybookjs/storybook/pull/12906/files#diff-a42d89f1ff025550a28baf1735d0dc7bd0feddea2fc8a576cf1e48d7ca7411ddR6

Reply:
Good catch @jimmyandrade, fixed in the main PR for this issue